gravitee-apim-console-webui/src/management/settings/portal-ng/portal-ng.component.ts

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, UntypedFormGroup } from '@angular/forms';
+import { combineLatest, Subject } from 'rxjs';
+import { takeUntil, tap } from 'rxjs/operators';
+
+import { PortalSettingsService } from '../../../services-ngx/portal-settings.service';
+import { SnackBarService } from '../../../services-ngx/snack-bar.service';
+import { PortalSettings } from '../../../entities/portal/portalSettings';
+import { GioPermissionService } from '../../../shared/components/gio-permission/gio-permission.service';
+import { CorsUtil } from '../../../shared/utils';
+
+@Component({
+  selector: 'portal-ng',
+  templateUrl: './portal-ng.component.html',
+  styleUrls: ['./portal-ng.component.scss'],
+})
+export class PortalNgComponent implements OnInit {
+  settings: PortalSettings;
+  portalForm: UntypedFormGroup;
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  public formInitialValues: unknown;
+  public defaultHttpHeaders = CorsUtil.defaultHttpHeaders.map((e) => e);
+  public httpMethods = CorsUtil.httpMethods;
+  public isLoadingData = true;
+  primaryOwnerModeList = [
+    {
+      id: 'HYBRID',
+      label: 'HYBRID: an API primary owner can be either a user or a group (Default)',
+    },
+    {
+      id: 'USER',
+      label: 'USER: an API primary owner can only be a user',
+    },
+    {
+      id: `GROUP`,
+      label: 'GROUP: an API primary owner can only be a group',
+    },
+  ];
+
+  constructor(
+    private readonly portalSettingsService: PortalSettingsService,
+    private readonly snackBarService: SnackBarService,
+    private readonly permissionService: GioPermissionService,
+  ) {}
+
+  public ngOnInit() {
+    this.isLoadingData = true;
+
+    combineLatest([this.portalSettingsService.get()])
+      .pipe(
+        tap(([portalSettings]) => {
+          this.settings = portalSettings;
+          this.initialPortalForm();
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => {
+        this.isLoadingData = false;
+      });
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  initialPortalForm() {
+    const settingsPermission = !this.permissionService.hasAnyMatching(['environment-settings-u']);
+
+    this.portalForm = new FormGroup({
```

Comment:
Instead of adding `settingsPermission` to each of the disabled, is there a way to disable the form as a whole?

Reply:
Done